src/Symfony/Component/Messenger/Transport/Sender/SendersLocator.php

```diff
@@ -41,20 +42,52 @@ public function __construct(array $sendersMap, ContainerInterface $sendersLocato
      */
     public function getSenders(Envelope $envelope): iterable
     {
-        $seen = [];
+        $senderAliases = [];
+
+        if (\PHP_VERSION_ID >= 80000) {
+            $senderAliases = $this->getSendersFromAttributes($envelope);
```

Comment:
should semantic config merge or overwrite?

Reply:
I merged them to keep transports from parent classes and interfaces. I'm not sure it will be obvious that the attribute will cancel the routing configured for its interface.

BTW, since I changed Senders to Transport - we don't have an option to configure behaviour with an argument on an attribute (because we can't have different behaviour for different transports of one class), but it can be done using another attribute (probably an overkill).